source/georef/astar_path_finder.cpp

```diff
@@ -49,10 +49,12 @@ void AstarPathFinder::init(const type::GeographicalCoord& start_coord,
     size_t n = boost::num_vertices(geo_ref.graph);
     costs.assign(n, bt::pos_infin);
 
-    costs[starting_edge[source_e]] =
-        compute_cost_from_starting_edge_to_dist(starting_edge[source_e], dest_projected_coord);
-    costs[starting_edge[target_e]] =
-        compute_cost_from_starting_edge_to_dist(starting_edge[target_e], dest_projected_coord);
+    if (starting_edge.found) {
```

Comment:
This is a good fix, but it doesn't prevent us from doing the same mistake ever again.....
I would be in favour of raising an exception in `ProjectionData::operator[] if `found` isn't set. 
Or you could change the way `ProjectionData::vertices` is implemented and make sure it's always in a state that will not bring you to Undefined Behavior Land.

Reply:
Just updated my PR to throw an exception in `ProjectionData::operator[]` when `found` is false ! :D